google-cloud-firestore/lib/google/cloud/firestore/watch/listener.rb

```diff
@@ -288,7 +291,8 @@ def background_run
           rescue RestartStream
             retry
           rescue StandardError => e
```

Comment:
An earlier rescue clause handles GRPC exceptions. Do we want to report those as well?

Reply:
Yes, at the point where it raises after hitting the backoff limit. Should the method return or raise after reporting the error?

Reply:
I'm updating this handler to report the error before it raises it, otherwise leaving the handling unchanged.